libs/execution_context/ebpf_native.c

```diff
@@ -1148,18 +1154,18 @@ ebpf_native_load(
             *module_id);
         goto Done;
     }
-    if (module->detaching || module->unloading) {
-        // This client is already detaching / unloading.
+    if (module->detaching || module->state == MODULE_STATE_UNLOADING) {
```

Comment:
I think we should merge the if check at line 1146 and line 1157 and set the result based on the current module state.

Reply:
(comment to same suggestion below)

Reply:
We can still set different return values based on the current module state. But will be better to merge the state check under one condition.

Also, per current PR changes, if the state is `MODULE_STATE_UNLOADING`, this function will bail from line 1155 and never hit line 1157.